src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/Whitelist.java

```diff
@@ -27,14 +27,50 @@
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
 import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
 
+import static java.lang.reflect.Modifier.isStatic;
+
 /**
  * Determines which methods and similar members which scripts may call.
  */
 public abstract class Whitelist {
 
+    private List<String> getEnvWhitelistRegex = new ArrayList<>();
+
+    public void setGetEnvWhitelistRegex(List<String> getEnvWhitelistRegex) {
+        this.getEnvWhitelistRegex = getEnvWhitelistRegex;
+    }
+
+    /**
+     * Return true if the given method is allowed System.getEnv()
+     * @param m the method
+     * @param args the method arguments
+     * @return true if allowed, false otherwise
+     */
+    public boolean isAllowedGetEnvSystemMethod(@Nonnull Method m, @Nonnull Object[] args) {
+        // Check if the method is "getenv" and it's a static method in the System class
+        if ("getenv".equals(m.getName()) &&
```

Comment:
Please consider caching the actual method in this class constructor:
```
Method getenvMethod = System.class.getMethod("getenv", String.class);
```
Then, in this method, you can use `equals` to check if it matches
`Method.equals` will check declaring class, method name and method params list.
We get here from ` onStaticCall` and System only exposes static methods anyway, so that should be enough.

Reply:
Updated